OpenRA.Game/Actor.cs

```diff
@@ -533,14 +533,14 @@ void UpdateConditionState(string condition, int token, bool isRevoke)
 		/// Otherwise, just returns InvalidConditionToken.
 		/// </summary>
 		/// <returns>The token that is used to revoke this condition.</returns>
-		public int GrantCondition(string condition)
+		public int Grant(GrantedVariableReference<ConditionType> condition)
```

Comment:
I'm not sure I like the idea of using such a generic name as just `Grant` here. Can we use `GrantVariable` if not `GrantCondition` here? In general though, I think it would be better if you could outline your vision for how the `Actor.Whatever` apis will look at the end of this work, and we can work through to some consensus on the naming.

Reply:
I plan on replacing it with what is currently `OpenRA.Actor.ConditionState` and renaming that `GrantCounting` and adding `IntegerSum`, possibly `GrantAny` and/or `MultiplierByPercentages`. Those would have say a `VariableAggregationBase` that defines how to take grants (possibly with an integer value) and reduce them to one value. `ConditionType` is just an attempt to avoid too much creeping in. `IntegerSum` would be for ammo and other places that push a number of tokens as well as prerequisite replacement. My WIP mock rules show why (https://github.com/atlimit8/OpenRA/tree/new-trait-expression-system-rules). I don't have everything I'm planning in there yet. If it is easier to review, I could do that in this PR. I'll try to get a couple of tracking issues open laying out my plans for changing the variable expression and trait systems.